### PR TITLE
Add op lowering for hstack and vstack.

### PR DESCRIPTION
**Pull request by zpcore**
Solves #5611

### CODE REVIEW

**Comment by zpcore** [comment]
Originally I tried with codegen, but it looks like `vstack` and `hstack` are not supported since I saw no file updates after I added those ops under full_codegen section in xla_native_functions.yaml.

File: test/cpp/test_aten_xla_tensor_1.cpp [hunk 1]
```diff
@@ -1470,6 +1470,34 @@ TEST_F(AtenXlaTensorTest, TestStack) {
   }
 }
 
+TEST_F(AtenXlaTensorTest, TestHstack) {
```
Comment (lsy323) [inline test/cpp/test_aten_xla_tensor_1.cpp:1473]: Can we also check the counter in the test, so we can make sure the op is lowered instead falling back to cpu

**Comment by lsy323** [comment]
> Originally I tried with codegen, but it looks like `vstack` and `hstack` are not supported since I saw no file updates after I added those ops under full_codegen section in xla_native_functions.yaml.

Weird, I saw hstack/vstack is in `native_functions.yaml` at https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L5517

File: test/cpp/test_aten_xla_tensor_1.cpp (hunk 1 above)
Comment (zpcore) [reply test/cpp/test_aten_xla_tensor_1.cpp:1473]: Good point. I was thinking the XLA op will automatically throws out the error if 1) the op lowering is not implemented and also 2) the input tensor is copied to the device. Looks like this is not the case. Thanks

**Comment by lsy323** [comment]
hstack/vstack is calling `cat` before dispatching happens. So lowering `cat` is enough for hstack/vstack

**Pull request closed without merging**